apps/BraynsViewer/main.cpp

```diff
@@ -26,18 +26,26 @@
 #include <brayns/parameters/ParametersManager.h>
 #include "BraynsViewer.h"
 
+brayns::Brayns* braynsInstance;
+
+void cleanup()
+{
+    delete braynsInstance;
+}
+
 int main(int argc, const char **argv)
 {
     try
     {
-        brayns::Brayns brayns( argc, argv );
+        braynsInstance = new brayns::Brayns( argc, argv );
         brayns::initGLUT( &argc, argv );
-        brayns::BraynsViewer braynsViewer( brayns );
+        brayns::BraynsViewer braynsViewer( *braynsInstance );
         BRAYNS_INFO << "Initializing Application..." << std::endl;
         const brayns::Vector2ui& size =
-            brayns.getParametersManager().getApplicationParameters().getWindowSize();
+            braynsInstance->getParametersManager().getApplicationParameters().getWindowSize();
 
         braynsViewer.create( "Brayns Viewer", size.x(), size.y( ));
+        atexit( cleanup );
```

Comment:
For doing and testing the destruction of engines and all other objects. Normally, you exit GLUT and the application with exit(), so immediate ending w/o any destruction, cleanup, resource freeing etc.